isabelle/Core/MultiAssets.thy

```diff
@@ -0,0 +1,262 @@
+theory MultiAssets
+imports Semantics 
+begin
+
+
+
+section "Assets"
+
+text "We represent Multi-token assets as a function from Token to natural numbers." 
+(*
+TODO: I want to replace Asset definition with 
+typedef Assets = "{assets. (\<forall> t v. fmlookup assets t = Some v \<longrightarrow> v > 0)} :: ((Token, nat) fmap) set
+
```

Comment:
Defining `Assets` as a `type_synonym` rather than a `typedef` should be fine. 
As we are based on `Finite_Map`, the type class instances might not be needed anymore

Reply:
We could try, but my guess is that this won't work as expected because I'm doing arithmetic with `Assets`

```
definition a1 = asset t1 4
definition a2 = asset t1 5
definition a3 = asset t2 6

definition result = a1 + a2 + a3 

```
which results in the new Asset ` (t1 -> 9, t2 ->6)`. In order to do that I need to define the `plus` instance for Assets, as you cannot sum `Finite_map`. Doing a newtype for `strictlyPositive` would require me to also build the same instances for that type as well.